docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+|BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) **storage**|?|
+|Amazing & Dazzling (A&D) Commands|Facade classes that deals with input so that different classes can interact with each other|
+|Parser|Parses the user input for command execution|
+
+#### 2.2.1 SAM record module
+
+The record module consists of 2 classes which represent the patients information and appointment details. 
+As a reflection of real world objects, the Patient's class purpose is to store the particulars of a person while the 
+Appointment's class is to store the date-time data. 
+Thus, the rationale of both classes can be grouped as follows:
+>
+> As a reflection of real world entities, to create, store, and retrieve relevant information about the object.
+>
+Following the above purpose, both classes consist of only getter and setter methods. This would ensure a contiguous 
+flow in logical executions as these methods can be called whenever necessary.
+
+##### 2.2.1.1 Process of Object Creation
+Due to the nature of the above classes containing only getter and setter methods, following how the components interact 
+with each other would provide more accuracy in understanding how these classes are called and the role of its 
+methods. 
+To illustrate, the below example is used:
+* editp \index 1 \name Justin \age 69 \Pasir Panjang
+
+![](images/SD_Patient.png)
+
+Upon startup, objects from ui, parser and storage are created. Prompted for user input, Duke receives the "editp"
+command which is forwarded to the parser to be interpreted respectively. Once the `EditPatientCommand` object is 
+created, it retrieves the patient index to edit the existing patient information from the patientList. 
+The `Patient` class is called by its setter method, `setPatientInfo()`, to update the fields as provided by the user. 
+This ensures that the encapsulated variables such as age, name, contact number and address are not only enforced but
+also protected. 
+
+##### 2.2.1.2 Design Considerations
+###### Aspect: Data Type for Appointment's Date and Time
+
++ Alternative 1 (current choice): Store as a String
+    * Pros: 
+        - Easier implementation
+        - Greater user flexibility
+    * Cons:
+        -  Cannot sort if needed
+
++ Alternative 2: Store as Date
+    * Pros:
+        - Has flexibility to parse or format date using existing methods available for use.
+    * Cons:
+        - Immutable-value classes mean it is not thread-safe (using Java.util.SimpleDateFormat).
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.2 Converter Module
+The converter module consists of one class which converts the format of date and time using a custom format defined by
+special formatting characters (ie. SimpleDateFormat). This class is primarily used to format a user-input date and time
+in the `Appointment` class. As illustrated below, its methods are called during the creation of the `Appointment`
+object constructor. 
+
+![](images/SD_Converter.png)
```

Comment:
![1](https://user-images.githubusercontent.com/47073395/77988754-abca6480-734f-11ea-90ee-944137ef1d7e.PNG)
For the activation bar of :TimeConverter, i think when object returns, the activation bar ends at the root return arrow.
What is the first Appointment object return arrow returning? 🤔